core/src/main/java/com/github/raulccabreu/redukt/middlewares/BaseAnnotatedMiddleware.kt

```diff
@@ -53,15 +53,29 @@ abstract class BaseAnnotatedMiddleware<T> : Middleware<T> {
     }
 
     private fun addBeforeActions(method: Method) {
+        val annotation = method.getAnnotation(BeforeActions::class.java) as BeforeActions
+
         verifyNumberOfArguments(method)
 
-        interceptBefores.add(method)
+        if (annotation.filter.isEmpty())
+            interceptBefores.add(method)
+        else
+            annotation.filter.forEach {
+                befores.put(it, method)
+            }
```

Comment:
A possible way to reduce duplication here.

```kotlin
private fun addBeforeActions(method: Method) {
  val annotation = method.getAnnotation(BeforeActions::class.java) as BeforeActions
  verifyNumberOfArguments(method)
  filtering(annotation, interceptBefores, befores, method)
}

private fun filtering(val annotation, val methods, val mapToMethod, val method) {
   if (annotation.filter.isEmpty()) methods.add(method)
   else annotation.filter.forEach { mapToMethod.put(it, method) }
}
```

Reply:
good solution.